dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/repositories/utils/TemporaryTables.kt

```diff
@@ -23,19 +23,27 @@ class TemporaryTables private constructor() {
             "FROM data_requests d " +
             "JOIN status_table ON status_table.request_id = d.data_request_id " +
             "WHERE " +
-            "(:#{#searchFilter.dataTypeFilterLength} = 0 " +
-            "OR d.data_type = :#{#searchFilter.dataTypeFilter}) AND " +
-            "(:#{#searchFilter.userIdFilterLength} = 0 " +
-            "OR d.user_Id = :#{#searchFilter.userIdFilter}) AND " +
-            "(:#{#searchFilter.requestStatusLength} = 0 " +
-            "OR status_table.request_status = :#{#searchFilter.requestStatus} ) AND " +
-            "(:#{#searchFilter.accessStatusLength} = 0 " +
-            "OR status_table.access_status = :#{#searchFilter.accessStatus}  ) AND " +
-            "(:#{#searchFilter.reportingPeriodFilterLength} = 0 " +
-            "OR d.reporting_period = :#{#searchFilter.reportingPeriodFilter}) AND " +
-            "(:#{#searchFilter.datalandCompanyIdFilterLength} = 0 " +
-            "OR d.dataland_company_id = :#{#searchFilter.datalandCompanyIdFilter}) " +
+            "(:#{#searchFilter.shouldFilterByDataType} = false " +
+            "OR d.data_type IN :#{#searchFilter.usedDataTypeFilter}) AND " +
+            "(:#{#searchFilter.shouldFilterByUserId} = false " +
+            "OR d.user_Id = :#{#searchFilter.usedUserIdFilter}) AND " +
+            "(:#{#searchFilter.shouldFilterByEmailAddress} = false " +
```

Comment:
Is the email address part of the table content? 
I thought the userId is stored in the table and we have to convert the specified emails into userIds.

Reply:
The email address is not stored in the table. The user_id is actually filtered twice in the query. Firstly, by the userId filter as before and as seen here. However, this filter is independent of the email filter and therefore remains unchanged. Second, by the email address filter in line 31, which filters the userIds associated with the email address.

Reply:
Ok then i would suggest to rename searchFilter.usedEmailAddressFilter to something else, to make it clear that these are also userIds. When i read it for the first time i thought it were actual e-mail addresses.
Maybe it is also possible to just merge the userId and the userIds from the Email search into one variable. This could be done in the Searchfilter. Then we would have one parameter less here.

Reply:
I can give the variable another name. But we can not merge the userId search with the email search, since they express different things. With the userId filter, I get all all requests by one user. If I search addtionally for an email address, and perform the SQL query user_id IN [userId, userIdFromEmail1, userIdFromEmail2, ...], I would obtain results which do not satisfy the userId filter.

Reply:
I have renamed the variable in the newest version.

Reply:
ok